Add radium, lodash to devDeps; react to peerDeps

coopy opened the pull request:
Fixes https://github.com/FormidableLabs/victory/issues/176

- [x] Update all subcomponent versions when they've been published

boygirl commented:
Approved!

austinpray commented:
Maybe lax up the react requirements to: `>=0.14.0` ?

coopy commented:
@austinpray We can't guarantee that Victory works with future React releases, though, can we?

knowbody commented:
this looks good to me! thanks @coopy 

@austinpray I don't think we can do that until future version of React is released. we support all 0.14.x which is good

austinpray commented:
You can't guarantee compatibility with 14+, but by doing this you are guaranteeing incompatibility. 

I would lax requirements and perhaps drop a warning in the console for versions greater than 14.

ryan-roemer commented:
@alexlande -- Can you comment on Radium's choice to not have a `peerDependencies` and just stick `react` in `devDependencies` instead? Thanks!

alexlande commented:
@ryan-roemer: Afraid I can't speak to it much. The change was made in this PR: https://github.com/FormidableLabs/radium/pull/242

I expect the reasoning was that at the time there was a lot of confusion going around re: `peerDependencies` (whether they were deprecated [they weren't], how they worked in npm3 vs npm2, etc), and that the benefits weren't clear. Radium opted to let users bring whatever React version they wanted so that you could work with beta versions and the like.

It's in the `devDependencies` because examples and tests use it, not as a hack to avoid using `peerDependencies`.

coopy commented:
@austinpray Thanks for the comment! I removed the `peerDependencies` requirement as it doesn't give much in terms of safety, and would likely just be a hindrance for folks trying out `react-beta` releases, etc.

I added some documentation instead: https://github.com/FormidableLabs/builder-victory-component/#peerdependencies
